### PR TITLE
Updated SVHN dataset(Allowing to use the full training set)

### PR DESCRIPTION
**Pull request by Coderx7**
I was recently trying to replicate a papers result on Pytorch, but noticed the current implementation only allows for using either `train `or `extra `splits for training, however nearly all papers that have experimented with `SVHN`, use the full dataset, i.e both the train and extra splits together. 
So I added a few changes to the original implementation that allows one to easily train using the full dataset. and thought it would be a good idea to share it with everyone. So here it is. 
The API is backward compatible (everything is the same tbh) so now one can easily do : 
```
import torchvision.datasets as dtset
...
if args.dataset == 'svhn':
     train_dir,val_dir = dtset.SVHN(args.data_path, split='download-all', download=True).create_dataset(args.outputdir)
     num_classes = 10
...
    train_dataset = dset.ImageFolder( train_dir, transforms.Compose([transforms.ToTensor() ]))
    train_loader = torch.utils.data.DataLoader(train_dataset,
        batch_size=args.batch_size, shuffle=True,
        num_workers=args.workers, pin_memory=True, sampler=None)

    test_dataset = dset.ImageFolder(val_dir, transforms.Compose([transforms.ToTensor() ]))
    test_loader = torch.utils.data.DataLoader(test_dataset,
        batch_size=args.batch_size, shuffle=False,
        num_workers=args.workers, pin_memory=True)

```
and utilize the whole training set and easily replicate any papers result on SVHN in Pytorch.

For those who have a huge amount of memory and prefer the in-memory method, they can do : 
```
import torchvision.datasets as dtset
if args.dataset == 'svhn':
    train_data = dtset.SVHN(args.data_path, split='train-extra', transform=train_transform, download=True)
    test_data = dtset.SVHN(args.data_path, split='test', transform=test_transform, download=True)
    num_classes = 10
...
    train_loader = torch.utils.data.DataLoader(train_data, batch_size=args.batch_size, shuffle=True,
                           num_workers=args.workers, pin_memory=True)
    test_loader = torch.utils.data.DataLoader(test_data, batch_size=args.batch_size, shuffle=False,
                          num_workers=args.workers, pin_memory=True)
```

Hope its useful.

Best Regards,
Hossein

### CODE REVIEW

**Comment by fmassa** [comment]
Hi,

Thanks for the PR!

Quick question: is there an advantage of this PR over using `torch.utils.data.dataset.ConcatDataset` with the current SVHN datasets?

**Comment by Coderx7** [comment]
Hi, 
No. The reason this is implemented is that I'm simply new to Pytorch and I didn't know about `torch.utils.data.dataset.ConcatDataset` at all!
Actually `torch.utils.data.dataset.ConcatDataset` is a much better idea, however since SVHN is one of the major datasets which many papers present their results based on the full dataset, it would make much more sense to have it readily available.

**Comment by fmassa** [comment]
Ok, thanks for your reply!

One more question: is `train+extra` an official split of the dataset, or was it done afterwards?

I'm a bit ambivalent on integrating this change given that it's fairly easy to combine datasets with `ConcatDataset`, and is also more robust (one can combine MNIST, SVHN, EMNIST, etc into a single dataset).

What do you think?

**Comment by Coderx7** [comment]
Yes train+extra is an official split of the dataset.
Yes, on the second thought it seems a simple example demonstrating the use of `torch.utils.data.dataset.ConcatDataset ` in the documentation can easily make up for this change and the issue is really solved!
